code/modules/admin/debug.dm

```diff
@@ -1346,3 +1346,21 @@ var/datum/flock/testflock
 
 
 	boutput(usr, "<span class='notice'>Returned: [!isnull(returnval) ? returnval : "null"]</span>")
+
+/proc/debugRemoveComponent(var/datum/target = null)
+	var/list/dc = target.datum_components
+	if(!dc)
+		boutput(usr, "<span class='notice'>No components present on [target].</span>")
+		return
+
+	var/list/comps = dc[/datum/component]
```

Comment:
indexing... by type?

Reply:
It sounds weird, but components are stored in a somewhat peculiar way.

For reference, for the sake of consistency I tried to be close to components' TransferComponents proc, which iterates through them in a similar way: https://github.com/goonstation/goonstation/blob/2e116d0e8df159dcd84e5cc7625d111b5223eaaf/code/datums/components/_component.dm#L541